### ISSUE TITLE
Failing to execute rubocop after gem installing it

### ISSUE THREAD

**Issue by vkatsikaros** [bbatsov/rubocop#4660]
I am new to ruby so I apologize if I am missing sth very basic. I install rubocop (with gem) but after installing successfully it fails to execute, even with a simple `-V` 

-------------

## Expected behavior

I expect `rubocop --version` to show its version

## Actual behavior

`rubocop --version` fails with error

## Steps to reproduce the problem

```
$ gem install rubocop
Fetching: parallel-1.12.0.gem (100%)
Successfully installed parallel-1.12.0
Fetching: unicode-display_width-1.3.0.gem (100%)
Successfully installed unicode-display_width-1.3.0
Fetching: ruby-progressbar-1.8.1.gem (100%)
Successfully installed ruby-progressbar-1.8.1
Fetching: powerpack-0.1.1.gem (100%)
Successfully installed powerpack-0.1.1
Fetching: ast-2.3.0.gem (100%)
Successfully installed ast-2.3.0
Fetching: parser-2.4.0.0.gem (100%)
Successfully installed parser-2.4.0.0
Fetching: rainbow-2.2.2.gem (100%)
Building native extensions.  This could take a while...
Successfully installed rainbow-2.2.2
Fetching: rubocop-0.49.1.gem (100%)
Successfully installed rubocop-0.49.1
Parsing documentation for parallel-1.12.0
Installing ri documentation for parallel-1.12.0
Parsing documentation for unicode-display_width-1.3.0
Installing ri documentation for unicode-display_width-1.3.0
Parsing documentation for ruby-progressbar-1.8.1
Installing ri documentation for ruby-progressbar-1.8.1
Parsing documentation for powerpack-0.1.1
Installing ri documentation for powerpack-0.1.1
Parsing documentation for ast-2.3.0
Installing ri documentation for ast-2.3.0
Parsing documentation for parser-2.4.0.0
Installing ri documentation for parser-2.4.0.0
Parsing documentation for rainbow-2.2.2
Installing ri documentation for rainbow-2.2.2
Parsing documentation for rubocop-0.49.1
Installing ri documentation for rubocop-0.49.1
Done installing documentation for parallel, unicode-display_width, ruby-progressbar, powerpack, ast, parser, rainbow, rubocop after 28 seconds
8 gems installed

$ rubocop -V
/home/vag/.rvm/gems/ruby-2.1.4/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:23:in `source': uninitialized constant Gem::Source (NameError)
	from /home/vag/.rvm/gems/ruby-2.1.4/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:65:in `extension_dir'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1774:in `gem_build_complete_path'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1988:in `missing_extensions?'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/stub_specification.rb:127:in `missing_extensions?'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/basic_specification.rb:64:in `contains_requirable_file?'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:64:in `block in require'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:63:in `each'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:63:in `find'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:63:in `require'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/source.rb:1:in `<top (required)>'
	from /home/vag/.rvm/gems/ruby-2.1.4/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:23:in `source'
	from /home/vag/.rvm/gems/ruby-2.1.4/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:35:in `full_gem_path'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/basic_specification.rb:138:in `block in full_require_paths'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/basic_specification.rb:137:in `map'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/basic_specification.rb:137:in `full_require_paths'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1372:in `add_self_to_load_path'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1285:in `activate'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1316:in `block in activate_dependencies'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1302:in `each'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1302:in `activate_dependencies'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1284:in `activate'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_gem.rb:60:in `block in gem'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_gem.rb:59:in `synchronize'
	from /home/vag/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_gem.rb:59:in `gem'
	from /home/vag/.rvm/gems/ruby-2.1.4/bin/rubocop:22:in `<main>'
	from /home/vag/.rvm/gems/ruby-2.1.4/bin/ruby_executable_hooks:15:in `eval'
	from /home/vag/.rvm/gems/ruby-2.1.4/bin/ruby_executable_hooks:15:in `<main>'

```

## RuboCop version

Can't see it, it fails but I expect it to be 0.49.1. Other relevant info:

```
$ ruby --version
ruby 2.1.4p265 (2014-10-27 revision 48166) [x86_64-linux]

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial


```

**Comment by mikegee** [comment]
Here I've tried to duplicate the problem in a docker container (to isolate this from my system gems and whatnot):

```
% docker run -it --rm ruby:2.1.4 bash
Unable to find image 'ruby:2.1.4' locally
2.1.4: Pulling from library/ruby
a3ed95caeb02: Pull complete
5d3df020ecd3: Pull complete
c09c5b1f622e: Pull complete
2f834e23e9c8: Pull complete
7fe16aa6941e: Pull complete
d9fd151ba672: Pull complete
92609fd9e7c7: Pull complete
9b2eb769553e: Pull complete
a52c5be7b122: Pull complete
Digest: sha256:88740a1f68a4d633abf82595374c038cedc8777fbe000fc79d212f590c4d8876
Status: Downloaded newer image for ruby:2.1.4
root@cbad9c0ee955:/# gem install rubocop
Fetching: rainbow-2.2.2.gem (100%)
Building native extensions.  This could take a while...
Successfully installed rainbow-2.2.2
Fetching: ast-2.3.0.gem (100%)
Successfully installed ast-2.3.0
Fetching: parser-2.4.0.0.gem (100%)
Successfully installed parser-2.4.0.0
Fetching: powerpack-0.1.1.gem (100%)
Successfully installed powerpack-0.1.1
Fetching: ruby-progressbar-1.8.1.gem (100%)
Successfully installed ruby-progressbar-1.8.1
Fetching: unicode-display_width-1.3.0.gem (100%)
Successfully installed unicode-display_width-1.3.0
Fetching: parallel-1.12.0.gem (100%)
Successfully installed parallel-1.12.0
Fetching: rubocop-0.49.1.gem (100%)
Successfully installed rubocop-0.49.1
8 gems installed
root@cbad9c0ee955:/# rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.1.4 x86_64-linux)
```

So I guess there's something misconfigured in your system, but I'm not sure what.

**Comment by pocke** [comment]
It looks failing in bundler. Could you tell me bundler and gem versions?

```bash
$ gem --version
$ bundler --version
```

I guess this problem is able to be resolved by updating those. 

```bash
$ gem update --system
$ gem update bundler
```

Could you try it?

**Comment by vkatsikaros** [comment]
@mikegee @pocke thanks for the comments

The current versions
```
$ gem --version
2.4.2
$ bundler --version
Bundler version 1.13.0
```
The updates
```
Updating rubygems-update
Fetching: rubygems-update-2.6.12.gem (100%)
Successfully installed rubygems-update-2.6.12
Parsing documentation for rubygems-update-2.6.12
Installing ri documentation for rubygems-update-2.6.12
Installing darkfish documentation for rubygems-update-2.6.12
Done installing documentation for rubygems-update after 36 seconds
Parsing documentation for rubygems-update-2.6.12
Done installing documentation for rubygems-update after 0 seconds
Installing RubyGems 2.6.12
RubyGems 2.6.12 installed
Parsing documentation for rubygems-2.6.12
Installing ri documentation for rubygems-2.6.12

=== 2.6.12 / 2017-04-30

Bug fixes:

...skip...
------------------------------------------------------------------------------

RubyGems installed the following executables:
	/home/vag/.rvm/rubies/ruby-2.1.4/bin/gem

Ruby Interactive (ri) documentation was installed. ri is kind of like man 
pages for ruby libraries. You may access it like this:
  ri Classname
  ri Classname.class_method
  ri Classname#instance_method
If you do not wish to install this documentation in the future, use the
--no-document flag, or set it as the default in your ~/.gemrc file. See
'gem help env' for details.

RubyGems system software updated


$ gem update bundler
Updating installed gems
Updating bundler
Fetching: bundler-1.15.3.gem (100%)
Successfully installed bundler-1.15.3
Parsing documentation for bundler-1.15.3
Installing ri documentation for bundler-1.15.3
Installing darkfish documentation for bundler-1.15.3
Done installing documentation for bundler after 8 seconds
Parsing documentation for bundler-1.15.3
Done installing documentation for bundler after 4 seconds
Gems updated: bundler
```
After the updates:
```
$ gem --version
2.6.12
$ bundler --version
Bundler version 1.15.3
```

And now rubocop works!!
```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.1.4 x86_64-linux)
```

**Comment by vkatsikaros** [comment]
@pocke the problem is now resolved so many thanks! I'd like to abuse the issue thread for a question mainly for my education. Apart from the generic "running the latest stable versions is _good_" and the fact that the error messages said many things in the end about `.../lib/bundler/...` (which I hadn't noticed) was there anything else that made you ask for the update?

**Comment by pocke** [comment]
I noticed three facts for the update.

- The stacktrace
- RuboCop works well in docker environment( https://github.com/bbatsov/rubocop/issues/4660#issuecomment-321917464) and CI.
- "running the latest stable versions is good"


The first fact says "it is bundler or rubygems problem.".
The second fact says "The environemnts are almost the same, but something is different."
So I guessed the cause may be difference of versions.
